elastic/kibana
# [Upgrade Assistant] Minimize reindex attributes used to create credential hash

Fixes https://github.com/elastic/kibana/issues/123616

It was determined that the reindexOp SO attributes were changing from the time the [credentials are initially set](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/routes/reindex_indices/reindex_handler.ts#L73) to when the [reindex worker requests them](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/lib/reindexing/worker.ts#L222). 

The diff was caused by the addition of the `openAndClose` parameter. I believe this is related to changes made in https://github.com/elastic/kibana/pull/59890 (originally implemented via https://github.com/elastic/kibana/pull/58890).

For now, I am excluding the `reindexOptions` from the credential hash. This seems like the least obtrusive change given the time constraints. I've opened https://github.com/elastic/kibana/issues/123752 as a follow-up issue to continue investigating and potentially consider updating the credential store with the updated reindexOptions as an alternative. 

### How to test
Steps to reproduce are listed in https://github.com/elastic/kibana/issues/123616. Note that Kibana must be stopped/restarted on the third step of the reindexing process -  "Reindex documents" - in order to reproduce.

With this change, you should be able to click "Resume reindexing" and the reindexing process will finish successfully.

There are some existing functional tests that [run through the reindexing flow](https://github.com/elastic/kibana/blob/7.17/x-pack/test/upgrade_assistant_integration/upgrade_assistant/reindexing.js), as well as [jest tests for the credential store](https://github.com/elastic/kibana/blob/7.17/x-pack/plugins/upgrade_assistant/server/lib/reindexing/credential_store.test.ts). I'm not sure if there's a good way to test this specific scenario by restarting Kibana.
